### PR TITLE
Emit event for intermediate generation

### PR DESCRIPTION
**Pull request by schrockn**
PR adds creation of an event for every intermediate generated in dagster and flows it to the dagit when executed in the context dagit.

This proposes that in the payload we communicate the name of the output, the "storage_mode" (which current would be IN_MEMORY, FILESYSTEM, or S3; i don't like the name) and the "storage_object_id" (again a name I don't like) for things like file paths and s3 paths.

### CODE REVIEW

File: python_modules/dagit/dagit_tests/test_graphql.py [hunk 1]
```diff
@@ -1740,6 +1753,48 @@ def test_basic_sync_execution():
...
+    logs = result.data['startPipelineExecution']['run']['logs']['nodes']
+    assert isinstance(logs, list)
+    assert has_event_of_type(logs, 'PipelineStartEvent')
+    assert has_event_of_type(logs, 'PipelineSuccessEvent')
+    assert not has_event_of_type(logs, 'PipelineFailureEvent')
+
+    assert first_event_of_type(logs, 'PipelineStartEvent')['level'] == 'INFO'
+
+    sum_solid_output = get_step_output_event(logs, 'sum_solid.transform')
+    assert sum_solid_output['step']['key'] == 'sum_solid.transform'
```
Comment (mgasner) [inline python_modules/dagit/dagit_tests/test_graphql.py]: can we also assert on the output's name here

**Comment by mgasner** [comment]
Rebase and rerun snapshots before merge

**Comment by schrockn** [comment]
spurious failure

**Comment by schrockn** [comment]
tracking issue here for airflow failures: https://github.com/dagster-io/dagster/issues/975

**Pull request merged**